src/main/scala/com/gu/subscriptions/cas/service/SubscriptionService.scala

```diff
@@ -0,0 +1,95 @@
+package com.gu.subscriptions.cas.service
+
+import com.gu.membership.zuora.ZuoraApiConfig
+import com.gu.membership.zuora.soap.Zuora._
+import com.gu.membership.zuora.soap._
+import com.gu.monitoring.ZuoraMetrics
+import com.gu.subscriptions.cas.config.Configuration
+import com.gu.subscriptions.cas.model.SubscriptionExpiration
+import com.gu.subscriptions.cas.service.utils.ScheduledTask
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class SubscriptionService(zuoraClient: ZuoraClient, knownProducts: List[String]) {
+  import spray.json._
+  import DefaultJsonProtocol._
+
+  case class SubsRequest(subscriberId: Option[String], password:String)
+  implicit val subsRequestFormat = jsonFormat2(SubsRequest)
+
+  def extractZuoraSubscriptionId(subsRequest: SubsRequest): Option[String] =
+    subsRequest.subscriberId.filter(_.startsWith("A-S"))
+
+  def extractSubsRequest(requestJson: String): SubsRequest = {
+    requestJson.parseJson.convertTo[SubsRequest]
+  }
+
+  val samePostcode: (String, String) => Boolean = {
+    val format: String => String = _.replaceAll("\\s+", "").toLowerCase
+
+    (postcodeA, postcodeB) => format(postcodeA) == format(postcodeB)
```

Comment:
The postcode is transported as `password` for legacy reasons, but the mobile UI requires the person's postcode. 
I do agree that it is useful to get a better understanding as to how often this information is filled incorrectly, but think that CloudWatch is a better place to store them because Sentry sends emails to dev team with each time an occurrence, which shouldn't be the case.